pkg/distributor/forwarding/forwarding.go

```diff
@@ -94,14 +109,69 @@ func NewForwarder(cfg Config, reg prometheus.Registerer, log log.Logger) Forward
 			Help:      "The client-side latency of requests to forward metrics made by the Distributor.",
 			Buckets:   []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 20, 30},
 		}),
+
+		grpcClientsGauge: promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+			Name: "cortex_distributor_forward_grpc_clients",
+			Help: "Number of gRPC clients used by Distributor forwarder.",
+		}),
+		grpcClientRequestDuration: promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "cortex_distributor_forward_grpc_client_request_duration_seconds",
+			Help:    "Time spend doing requests to forwarding target via gRPC.",
+			Buckets: prometheus.ExponentialBuckets(0.001, 4, 6),
+		}, []string{"operation", "status_code"}),
```

Comment:
The naming of this is a bit inconsistent with the above `requestLatencyHistogram` / `distributor_forward_requests_latency_seconds`. Should we pick one term between `latency` and `duration` and then stick to that one for both metrics?

Reply:
I will change the new metric to use latency.

Reply:
Actually, let me just reuse the same histogram, wdyt?

Reply:
> Actually, let me just reuse the same histogram, wdyt?

Oh, we can't, because of labels.